pkg/cmd/plugin/plugin_list.go

```diff
@@ -28,6 +31,18 @@ func newCmdPluginList(ctx api.Context) *cobra.Command {
 				enc := json.NewEncoder(ctx.Out())
 				enc.SetIndent("", "    ")
 				return enc.Encode(plugins)
+			} else if commands {
+				for _, plugin := range plugins {
+					for _, command := range plugin.Commands {
+						fmt.Fprintf(ctx.Out(), "%s ", command.Name)
+					}
+				}
+				return nil
+			} else if completionDirs {
+				for _, plugin := range plugins {
+					fmt.Fprintf(ctx.Out(), "%s ", plugin.CompletionDir())
```

Comment:
Any reason for using spaces instead of newlines? It seems more common for quiet outputs.

Reply:
`var=($(dcos plugin list --completion-dirs))` should work with newlines

Reply:
I did it this way because shellcheck complains about that syntax. There's more info here
https://github.com/koalaman/shellcheck/wiki/SC2207
and I wouldn't be against fixing it but then we do start to run into problems of doing things in a bash3 or bash4 way.

Though really I haven't been too worried about that so far and I've been running bash4...

Reply:
ok, spaces are fine here as it's a hidden/internal command, however we will most likely have newlines in other commands (eg. dynamic autocompletion with `dcos marathon app list -q`).

Reply:
This is a good point. How do we want to handle differences in bash3 vs bash4? I'm not even sure what all of them are but at least for this case maybe I can make a function that'll take in a command and return a list and have that handle the different methods based on version.

Reply:
> How do we want to handle differences in bash3 vs bash4?

They say `# For bash 3.x+`, so the code with IFS should also work on bash 4 I assume?

Otherwise, I don't think there is anything intrinsically wrong in using the `var=()` syntax, from what they say : 

> You are doing unquoted command expansion in an array. This will invoke the shell's sloppy word splitting and glob expansion.

If a command output has a well-known limited set of characters (eg. a list of UUIDs), this will never cause the problem they mention.